cfme/cloud/security_group.py

```diff
@@ -1,20 +1,268 @@
+from utils.wait import wait_for
 
 
-:var list_page: A :py:class:`cfme.web_ui.Region` object describing elements on the list page.
-:var details_page: A :py:class:`cfme.web_ui.Region` object describing elements on the detail page.
-"""
+class SecurityGroupsToolbar(View):
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Dropdown('Download')
 
-from cfme.web_ui import Region, SplitTable
+    view_selector = View.nested(ItemsToolBarViewSelector)
 
 
-# Page specific locators
-list_page = Region(
-    locators={
-        'security_group_table': SplitTable(header_data=('//div[@class="xhdr"]/table/tbody', 1),
-            body_data=('//div[@class="objbox"]/table/tbody', 1))
-    },
-    title='Security Groups')
+class SecurityGroupsDetailsToolbar(View):
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Button(title='Download summary in PDF format')
 
 
-details_page = Region(infoblock_type='detail')
+class SecurityGroupsDetailsAccordion(View):
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
+
+
+class SecurityGroupsQuadIconEntity(BaseQuadIconEntity):
+    pass
+
+
+class SecurityGroupsTileIconEntity(BaseTileIconEntity):
+    quad_icon = ParametrizedView.nested(SecurityGroupsQuadIconEntity)
+
+
+class SecurityGroupsListEntity(BaseListEntity):
+    pass
+
+
+class NonJSSecurityGroupsEntity(NonJSBaseEntity):
+    quad_entity = SecurityGroupsQuadIconEntity
+    list_entity = SecurityGroupsListEntity
+    tile_entity = SecurityGroupsTileIconEntity
+
+
+def SecurityGroupsEntity():  # noqa
+    """Temporary wrapper for Volume Entity during transition to JS based Entity """
+    return VersionPick({
+        Version.lowest(): NonJSSecurityGroupsEntity,
+        '5.9': JSBaseEntity,
+    })
+
+
+class SecurityGroupsEntities(BaseEntitiesView):
+    @property
+    def entity_class(self):
+        return SecurityGroupsEntity().pick(self.browser.product_version)
+
+
+class SecurityGroupsDetailsEntities(View):
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+    table = Table("//div[@id='list_grid']//table")
+    properties = SummaryTable(title='Properties')
+    relationships = SummaryTable(title='Relationships')
+    smart_management = SummaryTable(title='Smart Management')
+    firewall_rules = SummaryTable(title="Firewall Rules")
+
+
+class SecurityGroupsAddEntities(View):
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+    network_manager = BootstrapSelect(id='ems_id')
+    cloud_tenant = Dropdown(text='admin')
+
+
+class SecurityGroupsAddForm(View):
+    network_manager = BootstrapSelect(id='ems_id')
+    security_groups_name = TextInput(name='name')
+    security_groups_description = TextInput(name='description')
+    cloud_tenant = Select(name='cloud_tenant_id')
```

Comment:
I think this needs to be a ConditionalView, because it's dependant on the selection of the `network_manager` widget.  When the form is displayed, before anything is filled, this widget isn't visible (and isn't in the DOM).

Reply:
@mshriver It is selecting `network_manager` and then selecting `cloud_tenant`. So I think no need to have ConditionalView, right ..?

Reply:
It'll end up selecting based on the order of the dictionary passed to fill. There's no guarantee that network_manager will always be selected first. It either needs to be in `before_fill` method that's used by View.fill, or use a conditionalview so it will be a WidgetDescriptor until the referenced widget is filled.